source/extensions/filters/http/cache/cache_insert_queue.h

```diff
@@ -50,8 +54,7 @@ class CacheInsertQueue {
   Event::Dispatcher& dispatcher_;
   const InsertContextPtr insert_context_;
   const size_t low_watermark_bytes_, high_watermark_bytes_;
-  OptRef<Http::StreamEncoderFilterCallbacks> encoder_callbacks_;
-  AbortInsertCallback abort_callback_;
+  OptRef<InsertQueueCallbacks> callbacks_;
```

Comment:
`Anything*` is always simpler than `OptRef<Anything>`, so this comment is essentially "OptRef shouldn't exist".

The value is `OptRef` adds linter assertions that you don't use a potential `nullptr`.

Reply:
IIUC, you're saying that e.g.
`callbacks_->insertQueueUnderLowWatermark()` would hit a lint warning if it weren't guarded by
`if (callbacks_.has_value())`. Is that right? Thanks--I was wondering what OptRef's value was.

Regardless, I shouldn't've raised the question in this PR, because the code already had OptRef here--sorry.

Reply:
@toddmgreer : fyi OptRef also has these benefits
 * a pointer might be the start of an array, but OptRef cannot
 * a pointer might be owned by the object containing it, but OptRef cannot
 * a pointer might be intended to never be null, but OptRef directly conveys programmer intent that has_value==false is a normal expectation.

Thus its narrower use-case conveys user intent more precisely than pointers in those three cases.

I will also note that OptRef was introduced to wrap  verbose uses with `optional<std::reference<T>>` with terser code. The verbose form was somewhat common in Envoy before, as raw pointers were discouraged in the style guide, so OptRef was a cleaner way of expressing that. Later OptRef's impl was changed to a raw pointer, reducing memory overhead compared to `optional<std::ref...>`